### PR TITLE
[defaulttype] Template aliases for CRS matrices

### PR DESCRIPTION
**Pull request by alxbilger**
______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by damienmarchal** [comment]
Is this PR a fix for a breaks because of the merge of the NameDecoder refactoring ?

File: Sofa/framework/DefaultType/src/sofa/defaulttype/TemplatesAliases.cpp [hunk 1]
```diff
@@ -42,17 +43,18 @@ TemplateAliasesMap& getTemplateAliasesMap()
 
 bool TemplateAliases::addAlias(const std::string& name, const std::string& result, const bool doWarnUser)
 {
-	TemplateAliasesMap& templateAliases = getTemplateAliasesMap();
-	if (templateAliases.find(name) != templateAliases.end())
+    TemplateAliasesMap& templateAliases = getTemplateAliasesMap();
+    if (templateAliases.find(name) != templateAliases.end())
```
Comment (damienmarchal) [inline Sofa/framework/DefaultType/src/sofa/defaulttype/TemplatesAliases.cpp:47]: I would be better to emit the warning in the RegisterTemplateAlias constructor using the returned bool to detect registration failure.

Comment (alxbilger) [reply Sofa/framework/DefaultType/src/sofa/defaulttype/TemplatesAliases.cpp:47]: This part of the PR is just code formatting. Indentation was a mess. So I won't touch it in this PR.

**Comment by alxbilger** [comment]
> Is this PR a fix for a breaks because of the merge of the NameDecoder refactoring ?

No, it's to apply the idea to matrices that `Vec3` corresponds to `Vec3d` if `SReal` is a double, and `Vec3f` if it is a `float`.

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

**Pull request merged**